src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp

```diff
@@ -687,27 +690,29 @@ bool PerformanceImplicitGemmForwardV4R4Xdlops_Padded_Gemm::IsFastToBeUsedForTuni
 
         const float ratio = float(grid_size) / grid_size_max_blockwise_gemm;
 
-        if(grid_size_max_blockwise_gemm > 600)
+        const auto num_cu = ctx.GetStream().GetMaxComputeUnits();
```

Comment:
[Question] Ditto.

And another question: This fragment of code seems copied in several other solver. Can we get by with one copy?

Reply:
Yes, I will create a solver refactor PR to address that and a few other issues.

Reply:
Sounds good.